### PR TITLE
HW23 Models & Relations, DB_schema

### PR DESCRIPTION
**Pull request by DelDmc**

### CODE REVIEW

File: src/oldtimers/models.py [hunk 1]
```diff
@@ -1,3 +1,296 @@
...
+class BaseModel(models.Model):
+    class Meta:
+        abstract = True
+
+    create_datetime = models.DateTimeField(null=True, auto_now_add=True)
+    last_update = models.DateTimeField(null=True, auto_now=True)
+
+
+class Vehicle(BaseModel):
+    YEAR_OF_PRODUCTION_MIN_YEAR = 1900
```
Comment (mikolaz27) [inline src/oldtimers/models.py]: The word "YEAR" is twice here

File: src/oldtimers/models.py [hunk 2]
```diff
@@ -1,3 +1,296 @@
...
+        OTHER = 6, "Other"
+
+    owner = models.ForeignKey(
+        get_user_model(),
+        related_name="vehicle",
+        on_delete=models.CASCADE,
+        null=True,
+        blank=True,
+    )
+    retailer = models.ForeignKey(
```
Comment (mikolaz27) [inline src/oldtimers/models.py:79]: Looks like the retailer also needs access to the web site, so maybe you need to connect them with user model.

File: src/oldtimers/models.py [hunk 3]
```diff
@@ -1,3 +1,296 @@
...
+    )
+    condition = models.PositiveSmallIntegerField(
+        choices=VEHICLE_CONDITION_CHOICES.choices,
+        default=VEHICLE_CONDITION_CHOICES.OTHER,
+        null=True,
+        blank=True,
+    )
+    image = models.ImageField(
+        default="default.png",
+        upload_to="media/vehicle",
```
Comment (mikolaz27) [inline src/oldtimers/models.py:118]: add empty folder

File: src/oldtimers/models.py [hunk 4]
```diff
@@ -1,3 +1,296 @@
...
+        return self.vehicles.count()
+
+    def save(self, *args, **kwargs):
+        self.full_clean()
+        return super().save(*args, **kwargs)
+
+
+class Offer(BaseModel):
+    retailer = models.ForeignKey(to="oldtimers.Retailer", related_name="offers", on_delete=models.CASCADE)
+    vehicle = models.OneToOneField(
```
Comment (mikolaz27) [inline src/oldtimers/models.py:146]: looks like useless model.  Offer_price you can calculate on a "fly". In vehicle model, you have FK to the retailer.

File: src/oldtimers/models.py [hunk 5]
```diff
@@ -1,3 +1,296 @@
...
+        max_length=128,
+        null=False,
+        blank=True,
+    )
+
+
+class DeliveryService(BaseModel):
+    uuid = models.UUIDField(default=uuid.uuid4, db_index=True, unique=True)
+    company_name = models.CharField(max_length=50, null=False, blank=True)
+    mobile = PhoneNumberField(null=False, blank=True)
```
Comment (mikolaz27) [inline src/oldtimers/models.py]: phone_number - is better

**Comment by DelDmc** [comment]
closed due to some corrections

**Pull request closed without merging**